java-manta-client/src/test/java/com/joyent/manta/client/crypto/MantaEncryptedObjectInputStreamTest.java

```diff
@@ -932,9 +931,6 @@ private void canReadByteRange(SupportedCipherDetails cipherDetails,
              * the binary of the actual object (even if the range went beyond). */
             long ciphertextByteRangeLength = Math.min(ciphertextRangeMaxBytes, ciphertextRangeRequestBytes);
 
-            if (unboundedEnd)
-                ciphertextByteRangeLength = ciphertextSize - ciphertextStart; // include MAC
```

Comment:
This line seems to be attempting to prevent the situation you're describing, though `unboundedEnd` might not be the best name since it'll be true when `endPositionExclusive >= plaintextSize` (i.e. explicitly requesting an end that is beyond the plaintext size like the situation you describe in #245 )